gravitee-apim-cypress/cypress/integration/rest-api/apis/plan-tests.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ADMIN_USER, API_PUBLISHER_USER, LOW_PERMISSION_USER } from 'fixtures/fakers/users/users';
+import { ApiFakers } from 'fixtures/fakers/apis';
+import { PlanFakers } from 'fixtures/fakers/plans';
+import { createApi, deleteApi, stopApi } from 'commands/management/api-management-commands';
+import { createPlan, deletePlan, publishPlan } from 'commands/management/api-plan-management-commands';
+import { SecurityType, NewPlanEntity } from '@model/plan';
+
+context('Plan tests', () => {
+  describe('Create a Plan', function () {
+    beforeEach(function () {
+      const fakeApi = ApiFakers.api();
+      createApi(ADMIN_USER, fakeApi, true).its('body.id').as('apiId');
+    });
+
+    afterEach(function () {
+      deletePlan(ADMIN_USER, this.apiId, this.planId).its('status').should('equal', 204);
+      deleteApi(ADMIN_USER, this.apiId).its('status').should('equal', 204);
+    });
+
+    it('should create a keyless plan as admin user', function () {
+      const fakePlan = PlanFakers.plan({ security: SecurityType.KEY_LESS });
+      createPlan(ADMIN_USER, this.apiId, fakePlan).should(function (response) {
+        expect(response.status).to.equal(201);
+        expect(response.body).to.have.all.keys(
+          'name',
+          'id',
+          'description',
+          'validation',
+          'security',
+          'type',
+          'status',
+          'api',
+          'order',
+          'created_at',
+          'updated_at',
+          'paths',
+          'flows',
+          'comment_required',
+        );
+        expect(response.body.security).to.equal('KEY_LESS');
+        expect(response.body.status).to.equal('STAGING');
+        cy.wrap(response.body.id).as('planId');
+      });
+    });
+  });
+
+  describe('Publish plan', function () {
+    beforeEach(function () {
```

Comment:
```suggestion
    beforeEach('Create an API', () => {
```

Reply:
We'll need to define a coding convention here. My understanding is that Cypress is often used with regular functions (function () {}) since it makes use of the keyword this which changes its behaviour in Arrow functions. We'll need keep it in mind for a potential refactoring later on.